src/prototypes/draft-email-textarea/js/draft-email-textarea.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * The contents of this function will be inlined in a Gutenberg Custom HTML block
+ */
+export const runProposedInlineJS = () => {
+  // The elements all will be a part of the same Gutenberg Custom HTML block.
+  // I chose to use IDs since the HTML and JS are scoped to the same Gutenberg
+  // block (the JS functionaly won't be shared elsewhere, so I didn't use classes)
+  const copyBtn = document.getElementById('copy-btn');
+  const draftEmailBtn = document.getElementById('draft-email-btn');
+  const draftEl = document.getElementById('email-draft');
+  const copySuccessTooltipEl = document.getElementById('copy-success-tooltip');
+  const copyFailTooltipEl = document.getElementById('copy-fail-tooltip');
+
+  // All the things required to run this feature
+  const requirements = [
+    copyBtn,
+    draftEmailBtn,
+    draftEl,
+    copySuccessTooltipEl,
+    copyFailTooltipEl,
+    navigator,
+    navigator.clipboard,
+    navigator.clipboard.writeText,
+  ];
```

Comment:
Nit: this might be a little clearer:
```suggestion
  const requiredElements = [
    copyBtn,
    draftEmailBtn,
    draftEl,
    copySuccessTooltipEl,
    copyFailTooltipEl,
  ];

  const requiredBrowserFunctionality = [
    navigator,
    navigator.clipboard,
    navigator.clipboard.writeText,
  ]

  const requirements = [
    ...requiredElements,
    ...requiredBrowserFunctionality
  ];
```

(At first glance I skimmed this and assumed it was only checking for the elements. Later on I wondered how we were checking for the necessary functionality.)

Reply:
I like it!